docs/DeveloperGuide.md

```diff
@@ -1,38 +1,254 @@
 # Developer Guide
 
-## Acknowledgements
+## Introduction
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+Budget Buddy aims to help those who want a Command Line Interface App that is a one-stop solution to keep track of their
+expenses, help them with budgeting and also be help them to visualise what they have spent.
 
-## Design & implementation
+Overspending and the lack of savings is a significant problem found in most fresh graduates. We want to create tools
+necessary for them so that they can realize their actual spending and achieve financial freedom in the future.
+
+## Setting up the project
+
+## Design
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Main Component
+
+![MainSeq.png](uml/MainSequence.png)
```

Comment:
<img width="376" alt="Screenshot 2023-03-30 at 1 22 29 PM" src="https://user-images.githubusercontent.com/122626659/228737463-ab6cb485-dd0f-4816-bd2b-ad88d64df1bd.png">
The 'X' towards the bottom that indicates the end of a lifecycle should not overlap with the 'Box'